### DISCUSSION TITLE
Unable to resolve "@tanstack/react-query"

### DISCUSSION THREAD

**Discussion by SunDr17** [WalletConnect/walletconnect-monorepo#4146] [Developer Support]
Hello, I am trying to use Web3Modal following the docs:
https://docs.walletconnect.com/web3modal/react-native/about

but I am getting this error:

Unable to resolve "@tanstack/react-query" from "node_modules/wagmi/dist/esm/hooks/useBlockNumber.js"

I am trying to run web app. Not on mobile. I have the same codebase for all type of devices ios, android, web. So I have problems in web

Any one has solved this?

I am using expo 49

**Comment by glitch-txs** [comment]
install this versions wagmi@1.4.13 viem@1.21.4

**Reply by SunDr17** [reply]
yes, I used that versions

**Reply by SunDr17** [reply]
I tried to use this libs on empty expo project, run it on web as
`npm run web`
And I'am getting next errors
![2024-01-19 12 13 54](https://github.com/WalletConnect/walletconnect-monorepo/assets/12720040/c3c3fff2-ac4f-43f2-a868-2b233b8fb2bd)

**Reply by glitch-txs** [reply]
could you share your code?

**Reply by SunDr17** [reply]
Here my public repo: https://github.com/SunDr17/react-native-expo-web3/tree/master   (branch master)

My steps:
Node 20.11.0
### Installation
```shell
npx create-expo-app react-native-expo-web3 // create new empty expo project
npx expo install react-native-web react-dom @expo/metro-runtime // adding libs for running web
npx expo install @web3modal/wagmi-react-native wagmi@1.4.13 viem@1.21.4
npx expo install @react-native-async-storage/async-storage react-native-get-random-values react-native-svg react-native-modal @react-native-community/netinfo @walletconnect/react-native-compat
``` 
### Code changes
```typescript
// App.js
import '@walletconnect/react-native-compat'
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import { WagmiConfig } from 'wagmi'
import { mainnet, polygon, arbitrum } from 'viem/chains'
import { createWeb3Modal, defaultWagmiConfig, Web3Modal } from '@web3modal/wagmi-react-native'

// 1. Get projectId at https://cloud.walletconnect.com
const projectId = 'PROJECT_ID' // added real projectId

// 2. Create config
const metadata = {
  name: 'Web3Modal RN',
  description: 'Web3Modal RN Example',
  url: 'https://web3modal.com',
  icons: ['https://avatars.githubusercontent.com/u/37784886'],
  redirect: {
    native: 'YOUR_APP_SCHEME://',
    universal: 'YOUR_APP_UNIVERSAL_LINK.com'
  }
}

const chains = [mainnet, polygon, arbitrum]

const wagmiConfig = defaultWagmiConfig({ chains, projectId, metadata })

// 3. Create modal
createWeb3Modal({
  projectId,
  chains,
  wagmiConfig
})

export default function App() {
  return (
    <WagmiConfig config={wagmiConfig}>
      <View style={styles.container}>
        <Text>Open up App.js to start working on your app!</Text>
        <StatusBar style="auto" />
      </View>
      <Web3Modal />
    </WagmiConfig>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
``` 

### Running
```shell
npm run web
``` 
### Result
I am getting next errors:
<img width="927" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/12720040/4d68f5ad-7a67-48ec-82a8-b24df79c0182">


I tried to add 
```typescript
import React from 'react';
``` 
before and after `import '@walletconnect/react-native-compat'` and before and after `import { createWeb3Modal, defaultWagmiConfig, Web3Modal } from '@web3modal/wagmi-react-native'`
All tests failed with errors above

**Reply by SunDr17** [reply]
The same issue when I am running an ios app in simulator (`npm run ios`) 
`react-native-compat: Application module is not available`

**Reply by glitch-txs** [reply]
You also have react installed right?

**Reply by SunDr17** [reply]
Yes, you can see it here:
https://github.com/SunDr17/react-native-expo-web3/blob/master/package.json#L19
https://github.com/SunDr17/react-native-expo-web3/blob/master/package-lock.json#L14542

**Reply by glitch-txs** [reply]
Could you take a look? @ignaciosantise

**Reply by ignaciosantise** [reply]
hey @SunDr17 to solve the `Application module is not available` issue please install `expo-application` package in your project. I'll add this to our docs.

And about the "React is not defined" error, can you check [this](https://github.com/WalletConnect/modal-react-native/issues/30#issuecomment-1631083948)?

**Reply by SunDr17** [reply]
@ignaciosantise now i'am getting next error:
Unable to resolve "unstorage" from "node_modules/@walletconnect/keyvaluestorage/dist/index.es.js"

**Reply by ignaciosantise** [reply]
@SunDr17 are you getting this error on react native?

**Reply by SunDr17** [reply]
I am getting it in console when running web application. You can see all my steps, I described all flow in comment here. It just clear run of empty application, following your guide and running web app

**Reply by ignaciosantise** [reply]
@SunDr17 im not sure why are you getting that error, as "unstorage" is present in your lockfile. Anyways, you should know that web3modal RN is not meant to be used in web

**Reply by SunDr17** [reply]
and what do you suggest to use on web? I am creating universal app for web/ios/android

**Reply by ignaciosantise** [reply]
this is our web sdk -> https://docs.walletconnect.com/web3modal/react/about

You can check this project from Callstack that implements uses both sdks depending on the environment -> https://github.com/callstack/web3-cross-platform-dapp
